routes/main.js

```diff
@@ -12,38 +13,65 @@ exports.main = {
   config: {
     validate: {
       query: {
-        url: Joi.string().required(),
-        viewport: Joi.string().optional()
+        url: Joi.string().uri().required(),
+        viewport: Joi.string().optional(),
+        pixelRatio: Joi.number().integer().min(1).optional(),
+        wsEndpoint: Joi.string().optional()
       }
     }
   },
-  handler(request, reply) {
+  async handler(request, reply) {
     const server = request.server;
     const url = request.query.url;
     const randFldr = rando(18);
     const dir = `./screenshots/${randFldr}`;
+    const outputFile = `${dir}/screenshot.png`;
+    
+    try {
+      let browser;
 
-    const command = [];
+      fs.mkdirSync(dir);
```

Comment:
Just seen this, can we use `fs.mkdir` use `util.promisify` to make it `await`able.